private/model/api/api.go

```diff
@@ -631,6 +667,12 @@ func New(config aws.Config) *{{ .StructName }} {
     	),
     }
 
+	{{ if .HasExternalServiceConfigFields }}
+		if config.ExternalConfig != nil {
+			config.ExternalConfig.ResolveConfig(resolveServiceConfig(svc))
```

Comment:
even though not best place to use it, should capture error here, for when we add client builders.